csharp_dotnetcore/18.Multi-Lingual-Bot/Translation/MicrosoftTranslator.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using MultiLingualBot.Translation.Model;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace MultiLingualBot.Translation
+{
+    public interface ITranslator
+    {
+        Task<string> TranslateAsync(string text, string targetLocale);
+    }
+
+    public class MicrosoftTranslator : ITranslator
+    {
+        private const string host = "https://api.cognitive.microsofttranslator.com";
+        private const string path = "/translate?api-version=3.0";
+        private const string uriParams = "&to=";
+
+        private readonly string _key = "ENTER KEY HERE";
+
+        public MicrosoftTranslator(string key)
+        {
+            _key = key;
+        }
```

Comment:
Consider throwing something - with "Get a Key here" instructions - to prevent the sample from "almost" working.

Reply:
I'm removing this and actually validating this at startup time